### PR TITLE
Adding documentation for a Solace PubSub+ AMQP Component

### PR DESCRIPTION
**Pull request by TKTheTechie**
Thank you for helping make the Dapr documentation better!

**Please follow this checklist before submitting:**
- [ x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://docs.dapr.io/contributing/contributing-overview/#developer-certificate-of-origin-signing-your-work))
- [ x] [Read the contribution guide](https://docs.dapr.io/contributing/docs-contrib/contributing-docs/)
- [x] Commands include options for Linux, MacOS, and Windows within codetabs
- [x ] New file and folder names are globally unique
- [x] Page references use shortcodes instead of markdown or URL links
- [x] Images use HTML style and have alternative text
- [x] Places where multiple code/command options are given have codetabs

In addition, please fill out the following to help reviewers understand this pull request:

## Description

Adding documentation for a Solace PubSub+ AMQP Component

## Issue reference

[2382 ](https://github.com/dapr/components-contrib/pull/2383)(in components-contrib)

### CODE REVIEW

**Comment by msfussell** [comment]
@TKTheTechie - Given this PR https://github.com/dapr/components-contrib/pull/2383 is targetting the next v1.10 release, can you target this docs PR at the v1.10 docs repo instead of v1.9?

File: daprdocs/content/en/reference/components-reference/supported-pubsub/setup-solace-amqp.md [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+{{% /alert %}}
+
+## Spec metadata fields
+
+| Field              | Required | Details | Example |
+|--------------------|:--------:|---------|---------|
+| url    | Y  | Address of the AMQP broker. Can be `secretKeyRef` to use a secret reference. <br> Use the **`amqp://`** URI scheme for non-TLS communication. <br> Use the **`amqps://`** URI scheme for TLS communication. | `"amqp://host.domain[:port]"`
+| username | Y | The user name to connect to the broker with | default
+| password | Y | The password to connect to the broker with | default
+| anonymous | N | To connect to the broker without credential validation. Only works if enabled | true
```
Comment (msfussell) [inline daprdocs/content/en/reference/components-reference/supported-pubsub/setup-solace-amqp.md:42]: I presume that if this is set to `true` then you do not need username and password, is that correct. Either way can you make a comment here on how to use `anonymous` setting, correct and when?

File: daprdocs/content/en/reference/components-reference/supported-pubsub/setup-solace-amqp.md [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+      key: mySolaceClientKey
+auth:
+  secretStore: <SECRET_STORE_NAME>
+```
+
+Note that while the `caCert` and `clientCert` values may not be secrets, they can be referenced from a Dapr secret store as well for convenience.
+
+### Publishing/subscribing to topics and queues
+
+By default, messages are published and subscribed over topics. If you would like your destination to be a queue - prefix the topic with queue: and the Solace AMQP component will connect to a queue.
```
Comment (msfussell) [inline daprdocs/content/en/reference/components-reference/supported-pubsub/setup-solace-amqp.md:82]:
```suggestion
By default, messages are published and subscribed over topics. If you would like your destination to be a queue - prefix the topic with `queue:` and the Solace AMQP component will connect to a queue.
```

File: daprdocs/data/components/pubsub/generic.yaml [hunk 3]
```diff
@@ -2,49 +2,54 @@
...
   link: setup-rocketmq
   state: Alpha
   version: v1
-  since: "1.8"
+  since: '1.8'
+- component: Solace-amqp
+  link: setup-solace-amqp
+  state: Alpha
+  version: v1
+  since: '1.9'
```
Comment (msfussell) [inline daprdocs/data/components/pubsub/generic.yaml:55]:
```suggestion
  since: '1.10'
```

**Comment by TKTheTechie** [comment]
> 



> @TKTheTechie - Given this PR [dapr/components-contrib#2383](https://github.com/dapr/components-contrib/pull/2383) is targetting the next v1.10 release, can you target this docs PR at the v1.10 docs repo instead of v1.9?

Will do. Seems like I can't easily change the base due to conflicts. Will be easier to create a new PR.

**Pull request closed without merging**